tests/components/__init__.py

```diff
@@ -110,28 +110,43 @@ class StateDescription(TypedDict):
     """Test state and expected service call count."""
 
     state: str | None
+    attributes: dict
     count: int
 
 
 def parametrize_trigger_states(
-    trigger: str, target_states: Iterable[str], other_states: Iterable[str]
+    trigger: str,
+    target_states: Iterable[str | None | tuple[str | None, dict]],
```

Comment:
Can we make the item in the iterable a typed dict with all optional keys ("state", "attributes")? I think that would make the parameters more clear.